### PR TITLE
Ohm 368

### PR DESCRIPTION
**Pull request by BMartinos**
The PR looks up config variables for certificate management and determines if it should enable the file watcher. If the certificate/key is changed, the agenda task is kciked off to restart the server with the new certificate/key.

This PR depends on #812

### CODE REVIEW

**Comment by BMartinos** [comment]
Hey @rcrichton ok cool no problem, I think this will make things a lot easier and we wouldnt need to add agenda tasks then. This should be a quick change as well. Will get started on it.

**Comment by rcrichton** [comment]
@BMartinos cool, if you could write a test to ensure that the server is restarted when the cert changes that would be great to.

File: test/integration/e2eIntegrationTests.coffee [hunk 1]
```diff
@@ -1369,3 +1369,34 @@ describe "e2e Integration Tests", ->
...
+            fs.createReadStream('test/resources/certificate-watcher/testBackup.crt').pipe(fs.createWriteStream('test/resources/certificate-watcher/test.crt'));
+
+            Keystore.findOne {}, (err, keystore) ->
+              done(err) if err
+              keystore.cert.data.should.be.exactly 'New certificate content. should trigger certificate reload'
+              done()
+            return
+          ), 1500
+        
+
```
Comment (rcrichton) [inline test/integration/e2eIntegrationTests.coffee]: As we chatted about, remove the test until we can figure out a good way to test this... :/

File: src/server.coffee [hunk 2]
```diff
@@ -62,6 +63,22 @@ autoRetry = require './autoRetry'
 
 clusterArg = nconf.get 'cluster'
 
+exports.setupCertificateWatcher = () ->
+  certFile = config.certificateManagement.certPath
+  keyFile = config.certificateManagement.keyPath
+  watcher = chokidar.watch([certFile, keyFile], {
+    usePolling: true
```
Comment (rcrichton) [inline src/server.coffee]: Ok cool.

**Pull request merged**